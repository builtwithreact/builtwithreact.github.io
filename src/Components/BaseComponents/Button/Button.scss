.Button {
  color: #fff;
  font-size: 18px;
  border: 1px #fff solid;
  border-radius: 2px;
  padding: 10px;
  transition: all .3s ease;

  &:hover {
    background: #4caf4f;
    border-color: #4caf4f;
  }
}