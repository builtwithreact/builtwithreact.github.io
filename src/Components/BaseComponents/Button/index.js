import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Button extends Component {
  render() {
    return (
      <button type={this.props.type}
        className={this.props.className.trim()}
        onClick={this.props.handleOnClick}> 
        {this.props.value}
      </button>
    )
  }
}

Button.propTypes = {
  type: PropTypes.oneOf(['button', 'submit', 'reset']),
  value: PropTypes.string.isRequired,
  className: PropTypes.string,
  handleOnClick: PropTypes.func.isRequired
};

Button.defaultProps = {
  type: 'button',
  className: ""
}
export default Button;
