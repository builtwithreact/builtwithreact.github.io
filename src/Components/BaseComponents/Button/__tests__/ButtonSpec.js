import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Button from './../index';

describe('Button', ()=>{
    it('renders correctly if only mandatory props are passed', ()=>{
        let tree = renderer.create(
            <Button value="btn value" 
                handleOnClick={jest.fn} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly if options props are also passed', ()=>{
        let tree = renderer.create(
            <Button value="btn value"
                className="btnclass"
                handleOnClick={jest.fn} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should call passed click handler', ()=>{
        let mockHandOnClick = jest.fn();
        const wrapper = shallow( 
            <Button value="btn value"
                className="btnclass"
                handleOnClick={mockHandOnClick} />
        )
        wrapper.simulate('click');
        expect(mockHandOnClick).toBeCalled;
    });

});
