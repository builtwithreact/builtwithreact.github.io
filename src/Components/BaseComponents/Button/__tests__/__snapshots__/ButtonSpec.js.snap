// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly if only mandatory props are passed 1`] = `
<button
  className=""
  onClick={[Function]}
  type="button"
>
  btn value
</button>
`;

exports[`Button renders correctly if options props are also passed 1`] = `
<button
  className="btnclass"
  onClick={[Function]}
  type="button"
>
  btn value
</button>
`;
